src/components/Todo/TodoItem.tsx

```diff
@@ -0,0 +1,210 @@
+
+type Props = {
+  todo: Todo,
+  setErrorMessage: (str: string) => void,
+  loadingByDefault?: boolean,
+  focusMainInput: () => void,
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  setErrorMessage,
+  loadingByDefault = false,
+  focusMainInput,
+}) => {
+  const { title, completed, id } = todo;
+
+  const reducer = useContext(DispatchContext);
+
+  const [isChecked, setIsChecked] = useState(completed);
+  const [isEditing, setIsEditing] = useState(false);
+  const [isLoading, setIsLoading] = useState(loadingByDefault);
+  const [editedTitle, setEditedTitle] = useState(title);
+
+  const handleSetCompleted = () => {
+    setIsLoading(true);
+
+    updateTodo(id, { completed: !isChecked, userId: 11562 })
+      .then(() => {
+        setIsChecked(!isChecked);
+        reducer({ type: ActionType.SetCompleted, payload: id });
+      })
+      .catch(() => {
+        setErrorMessage('Unable to update a todo');
+
+        setTimeout(() => {
+          setErrorMessage('');
+        }, 3000);
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  };
+
+  const handleDeleteClick = () => {
+    setIsLoading(true);
+
+    deleteTodo(id)
+      .then(() => {
+        reducer({ type: ActionType.Delete, payload: id });
+        focusMainInput();
+      })
+      .catch(() => {
+        if (!isEditing) {
+          focusMainInput();
+        }
+
+        setErrorMessage('Unable to delete a todo');
+
+        setTimeout(() => {
+          setErrorMessage('');
+        }, 3000);
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  };
+
+  const handleDoubleClick = () => {
+    setIsEditing(true);
+  };
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setEditedTitle(event.target.value);
+  };
+
+  const handleFormSubmit = () => {
+    setIsLoading(true);
+
+    updateTodo(id, { title: editedTitle.trim(), userId: 11562 })
+      .then(() => {
+        setEditedTitle(editedTitle.trim());
+        reducer({
+          type: ActionType.EditTitle,
+          payload: { id, title: editedTitle.trim() }
+        });
+        setIsEditing(false);
+      })
+      .catch(() => {
+        setErrorMessage('Unable to update a todo');
+
+        setTimeout(() => {
+          setErrorMessage('');
+        }, 3000);
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  };
+
+  const handleBlur = () => {
+    if (editedTitle.trim() === '') {
+      handleDeleteClick();
+    } else {
+      if (editedTitle === title) {
+        setIsEditing(false);
+
+        return;
+      }
+
+      handleFormSubmit();
+    }
+  };
+
+  const handleKeyUp = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    event.preventDefault();
+
+    if (event.key === 'Enter') {
+      handleBlur();
+    } else if (event.key === 'Escape') {
+      setIsEditing(false);
+      setEditedTitle(title);
+    }
+  };
+
+  useEffect(() => {
+    setIsChecked(completed);
+  }, [todo]);
```

Comment:
![image](https://github.com/mate-academy/react_todo-app-with-api/assets/56752562/43b3dbf0-0e38-472c-ba08-6f5381cc190c)
Try to fix linter error